### ISSUE TITLE
Configure editable cells individually

### ISSUE THREAD

**Issue by step307** [wenzhixin/bootstrap-table#1789]
Literally it will mean, that the editable parameter should not only accept boolean or object, but also a function which returns boolean or an object.

This callback receive usual parameters like value, row, index.

**Comment by dabros** [comment]
@step307 - http://issues.wenzhixin.net.cn/bootstrap-table/#extensions/editable.html

That is individual editable cells, so far as i can tell

**Comment by step307** [comment]
@dabros , this is example of how to set editable for a whole column (similar for all rows). My feature request is to make it also definable per individual cell depending on cells's data (row).

It terms of this example the user story could be "Make editable only prices of items with id > 5"

**Comment by dabros** [comment]
@step307  - thanks for clarification

So very clear to all others, this is example of editable plugin table:

```

    <div class="container">
        <h1>Editable</h1>
        <table id="table"
               data-toggle="table"
               data-pagination="true"
               data-show-export="true"
               data-url="../json/data1.json">
            <thead>
            <tr>
                <th data-field="id">ID</th>
                <th data-field="name" data-editable="true">Item Name</th>
                <th data-field="price" data-editable="true">Item Price</th>
            </tr>
            </thead>
        </table>
    </div>
```

The plugin doco refered to from http://bootstrap-table.wenzhixin.net.cn/extensions/ is currently located at https://github.com/wenzhixin/bootstrap-table/tree/master/src/extensions/editable


> Column options
> 
> editable
> 
> type: Object
> description: Configuration of x-editable. Full list of options: http://vitalets.github.io/x-editable/docs.html#editable
> default: undefined



That to me reads that you 'can' use `editable` as simple bool, but that you can set it in js as an x-editable equatable object.

That appears reinforced by the calls to `editable` in https://github.com/wenzhixin/bootstrap-table/blob/master/src/extensions/editable/bootstrap-table-editable.js

```
that.$body.find('a[data-name="' + column.field + '"]').editable(column.editable)
```

Only way to confirm is to create fiddle with this and set via js.

No time atm sorry, do post if you start though else i might get round to it later.

**Comment by wenzhixin** [comment]
:+1: @dabros

**Comment by KonstantinObuhov** [comment]
This is my way for same task.
```
$table.on('load-success.bs.table', function(data) {
    $tableRows = $table.find('tbody tr');
    tableData = $table.bootstrapTable('getData', true);
    $.each(tableData, function (i, row) {
        if(2 == row.check_status) {
            $tableRows.eq(i).find('a').editable('toggleDisabled');
        }
    });
});
```

**Comment by Jingkun-Hu** [comment]
This is work version for me
 function unit_table_editable_check() {
            var i = 0;
            $("#your_table tr").each(function () {
                if (i == 0) {
                    i = 1;
                }
                else {
                    var floorType = $("td:nth-child(6)", this)["0"].innerText;
                    if (floorType == "col 7&12 is should be disabled") {
                        console.log($('td:nth-child(7)', this)["0"].childNodes["0"]);
                        $($('td:nth-child(7)', this)["0"].childNodes["0"]).off('click');
                        $($('td:nth-child(7)', this)["0"].childNodes["0"]).css("color", 'your color');
                        $($('td:nth-child(12)', this)["0"].childNodes["0"]).off('click');
                        $($('td:nth-child(12)', this)["0"].childNodes["0"]).css("color", 'your color');
                    }
                }
            });
        };

**Comment by wenzhixin** [comment]
Fixed in #2038
